### ISSUE TITLE
Debugging in Libraries in LeanCLI

### ISSUE THREAD

**Issue by jaredbroad** [QuantConnect/lean-cli#284]
Community contributor fixed library debugging on his environment using the fix below.. Should be made defaults.
```
The trick is to add
"sourceFileMap": {
"/Library": "${workspaceRoot}/Library"
}
to the root launch.json. Moving the launch.json to root and adding this should be defaults for the Lean CLI imo.
```

**Comment by connorjohnlind** [comment]
TLDR: the `launch.json` example listed at the end of this comment needs to be added in the root `.vscode` directory on `lean init`.

This fix was specifically for C# projects in VSCode. The [docs here](https://www.quantconnect.com/docs/v2/lean-cli/backtesting/debugging#06-C-and-VS-Code) are out of date. The debugger will not attach properly.

For example:

```
lean init
lean create-project Test --language csharp
```

To get the debugger to attach, I had to move `Test/.vscode/launch.json` to the root `.vscode/launch.json`. IMO this file should be created by default on `lean init`, before any projects are created.

Furthermore, libraries created by the CLI will not be found.
```
lean create-project Library/MyLibrary --language csharp
lean library add Test Library/MyLibrary
lean backtest Test --debug vsdbg    
```
The VSCode Debug Console will show the Library's symbols are found if `launch.json` has `"moduleLoad": true`. However, stepping into a Library function will show an "The editor could not be opened because the file was not found." error:

![Screen Shot 2023-02-19 at 20 12 34](https://user-images.githubusercontent.com/8453419/219969965-b6e77f7f-51cb-4f81-b448-023cc07759af.png)

I eventually discovered there was a hiccup in the path resolution. If adding the following to the root `.vscode/launch.json`, you are able to step into the library and set breakpoints in them:
```      
"sourceFileMap": {
    "/Library": "${workspaceRoot}/Library"
}
```
![Screen Shot 2023-02-19 at 20 19 36](https://user-images.githubusercontent.com/8453419/219970277-12f003bc-0c0b-4a8e-a4b1-a7f9514f3799.png)

The final `.vscode/launch.json` file is as follows:
```
{
  "configurations": [
    {
      "name": "Lean CLI",
      "request": "attach",
      "type": "coreclr",
      "processId": "1",
      "pipeTransport": {
        "pipeCwd": "${workspaceRoot}",
        "pipeProgram": "docker",
        "pipeArgs": ["exec", "-i", "lean_cli_vsdbg"],
        "debuggerPath": "/root/vsdbg/vsdbg",
        "quoteArgs": false
      },
      "logging": {
        "moduleLoad": false
      },
      "sourceFileMap": {
        "/Library": "${workspaceRoot}/Library"
      }
    }
  ]
}
```

**Comment by connorjohnlind** [comment]
I noticed that the [docs here](https://www.quantconnect.com/docs/v2/lean-cli/backtesting/debugging#06-C-and-VS-Code) do specify opening the debugger from the project directory, not the workspace. This does not work as stated, perhaps another CLI option is needed to run the debugger from the project directories:
```
% cd Test
% lean backtest Test --debug vsdbg

Usage: lean backtest [OPTIONS] PROJECT

Try 'lean backtest --help' for help or go to the following url for a list of common errors:
https://www.lean.io/docs/v2/lean-cli/key-concepts/troubleshooting#02-Common-Errors

Error: Invalid value for 'PROJECT': Path 'Test' does not exist.
```
 [This youtube video](https://www.youtube.com/watch?v=oRpJEHX676Y) also doesn't note this step and should get updated to reflect any new changes.
